src/main.zig

```diff
@@ -197,13 +219,43 @@ pub fn Deserializer(comptime ReaderType: type) type {
                 else => return error.MismatchingFormatType,
             };
             var value: T = undefined;
+            // A static array of bits (0 or 1) with the same len as the number of fields on the struct.
+            // If the value of array[i] is 1, it means that field at position i has been set and should be
+            // freed in case of an error or in case the map has duplicate keys
+            var fields_set = [_]u1{0} ** meta.fields(T).len;
```

Comment:
An array of `u1` will be padded to be equal to an array of `u8`, making it no more efficient. Personally, I would just use an array of booleans. While this takes just as much memory, it makes the logic easier to read instead of having to check for `1` to find out if a field is set or not.

Reply:
Yes good point. On the other hand, do you think it would be better to use a `std.StaticBitSet` instead of a bool array?

Reply:
I don't think the added complexity of using that warrants the stack memory savings. It could potentially be slower to write/read to/from while barely saving any cost as structs tend to not have too many fields to notice any stack size differences.

Reply:
Ok I will change it to bool arrays then 😃